pkg/network/vmispec/interface.go

```diff
@@ -58,6 +58,22 @@ func IsPodNetworkWithMasqueradeBindingInterface(networks []v1.Network, ifaces []
 	return true
 }
 
+func ShiftPrimaryInterface(statusIfaces []v1.VirtualMachineInstanceNetworkInterface, networks []v1.Network) []v1.VirtualMachineInstanceNetworkInterface {
+	if podNetwork := lookupPodNetwork(networks); podNetwork != nil {
+		for index, currStatusIface := range statusIfaces {
+			if currStatusIface.Name == podNetwork.Name {
+				if index > 0 {
+					return append(append([]v1.VirtualMachineInstanceNetworkInterface{statusIfaces[index]},
+						statusIfaces[:index]...), statusIfaces[index+1:]...)
+				} else {
+					break
+				}
+			}
+		}
+	}
+	return statusIfaces
```

Comment:
One suggestion that I would have is to reduce nestiness a little bit. The code itself is a little bit longer but I think it's easier to reason about, I think.

```suggestion
	podNetwork := lookupPodNetwork(networks)
	if podNetwork == nil {
		return statusIfaces
	}
	if len(statusIfaces) > 0 && statusIfaces[0].Name == podNetwork.Name {
		return statusIfaces
	}	
	for index, currStatusIface := range statusIfaces {
 		if currStatusIface.Name == podNetwork.Name {
 	 		return append(append([]v1.VirtualMachineInstanceNetworkInterface{statusIfaces[index]},
 						statusIfaces[:index]...), statusIfaces[index+1:]...)
		}
	}
	return statusIfaces
```

Reply:
I see your point. I usually try to minimize the return statements but sometimes it makes the code less readable. 
I'll change it.